lib/api/include/dataObjChksum.h

```diff
@@ -1,29 +1,116 @@
 #ifndef DATA_OBJ_CHKSUM_H__
 #define DATA_OBJ_CHKSUM_H__
 
-#include "rcConnect.h"
-#include "dataObjInpOut.h"
-#include "objInfo.h"
+/// \file
+
+struct RcComm;
+struct DataObjInp;
 
 #ifdef __cplusplus
-extern "C"
+extern "C" {
 #endif
-int rcDataObjChksum( rcComm_t *conn, dataObjInp_t *dataObjChksumInp, char **outChksum );
 
-/* rcDataObjChksum - Chksum a iRODS data object.
- * Input -
- *   rcComm_t *conn - The client connection handle.
- *   dataObjInp_t *dataObjInp - generic dataObj input. Relevant items are:
- *	objPath - the path of the data object.
- *	openFlags - should be set to O_RDONLY.
- *   condInput - FORCE_CHKSUM_KW - force checksum even if one exist.
- *		 CHKSUM_ALL_KW - checksum all replica.
- *		 VERIFY_CHKSUM_KW - verify the checksum value in icat.
- * 		 REPL_NUM_KW - the "value" gives the replica number of the
- *		   copy to checksum.
- * OutPut -
- *   char **outChksum - the chksum string
- *   return value - The status of the operation.
- */
+/// Computes and stores the checksum for a data object.
+///
+/// The default checksum algorithm is SHA256.
```

Comment:
hmmm...  could be a tad more descriptive/correct.

a link to the github repo (which has a branch in the URL)...
 - https://github.com/irods/irods_docs/blob/4-2-stable/docs/system_overview/configuration.md#checksum-configuration

or

a link to the rendered docs site (which has a version number in the URL)...
 - https://docs.irods.org/4.2.8/system_overview/configuration/#checksum-configuration

or

"The checksum algorithm used is determined by considering both the sender and receiver's hash schemes and the receiver's policy."

Reply:
all good points/concerns.    how about...

"The default checksum algorithm for iRODS is SHA256, but the actual algorithm used by this function is determined at runtime by the client and server's configuration."

Reply:
that is much better imo. i think the statement needs to be swapped though. how do you feel about this:

**The checksum algorithm used by this function is determined by the client and server's configuration, but defaults to SHA256.**

lastly, does this information belong in this function's documentation?

Reply:
yours is better, definitely.

I think it's... helpful to be here.  How about...

note the 'a' in the first sentence.
```
Computes and stores a checksum for a data object. 

The checksum algorithm used by this function is determined by the client and server's configuration, but defaults to SHA256.
```

Reply:
maybe "and" should replace "but" in my swap example.

Reply:
i'll use your first suggestion. the more i read it, i like it.

Reply:
updated based on your comment.

thanks sir!

Reply:
i think 'but' is fine.